node/full_node_integration_test.go

```diff
@@ -172,33 +172,39 @@ func TestLazyAggregator(t *testing.T) {
 
 	require.NoError(err)
 
-	require.NoError(waitForFirstBlock(node.(*FullNode)))
+	require.NoError(waitForFirstBlock(node.(*FullNode), false))
 
 	client := node.GetClient()
 
 	_, err = client.BroadcastTxCommit(context.Background(), []byte{0, 0, 0, 1})
 	assert.NoError(err)
-	require.NoError(waitForAtLeastNBlocks(node, 2))
+	require.NoError(waitForAtLeastNBlocks(node, 2, false))
 
 	_, err = client.BroadcastTxCommit(context.Background(), []byte{0, 0, 0, 2})
 	assert.NoError(err)
-	require.NoError(waitForAtLeastNBlocks(node, 3))
+	require.NoError(waitForAtLeastNBlocks(node, 3, false))
 
 	_, err = client.BroadcastTxCommit(context.Background(), []byte{0, 0, 0, 3})
 	assert.NoError(err)
 
-	require.NoError(waitForAtLeastNBlocks(node, 4))
+	require.NoError(waitForAtLeastNBlocks(node, 4, false))
 
 }
 
+func TestBlockExchange(t *testing.T) {
+	testSingleAggregatorSingleFullNode(t, true)
+	testSingleAggregatorTwoFullNode(t, true)
+	testSingleAggregatorSingleFullNodeTrustedHash(t, true)
+}
+
 func TestHeaderExchange(t *testing.T) {
-	testSingleAggreatorSingleFullNode(t)
-	testSingleAggreatorTwoFullNode(t)
-	testSingleAggreatorSingleFullNodeTrustedHash(t)
-	testSingleAggreatorSingleFullNodeSingleLightNode(t)
+	testSingleAggregatorSingleFullNode(t, false)
+	testSingleAggregatorTwoFullNode(t, false)
+	testSingleAggregatorSingleFullNodeTrustedHash(t, false)
+	testSingleAggregatorSingleFullNodeSingleLightNode(t)
```

Comment:
What makes this test different that we don't use the `useBlockExchange` bool?

Reply:
i think the light node does not have block exchange service, it only has header exchange.